packages/zoe/src/contracts/priceAggregatorChainlink.js

```diff
@@ -0,0 +1,743 @@
+        }
+
+        if (!(error.length === 0)) {
+          console.error(error);
+          return;
+        }
+
+        oracleInitializeNewRound(roundId, oracleInstance);
+        const recorded = recordSubmission(
+          parsedSubmission,
+          roundId,
+          oracleInstance,
+        );
+        if (!recorded) {
+          return;
+        }
+
+        updateRoundAnswer(roundId);
+        deleteRoundDetails(roundId);
+      };
+
+      // Obtain the oracle's publicFacet.
+      await E(zoe).getPublicFacet(oracleInstance);
+      assert(records.has(record), X`Oracle record is already deleted`);
+
+      /** @type {OracleAdmin} */
+      const oracleAdmin = {
+        async delete() {
+          assert(records.has(record), X`Oracle record is already deleted`);
+
+          // The actual deletion is synchronous.
+          oracleStatuses.delete(oracleInstance);
+          records.delete(record);
+
+          if (
+            records.size === 0 &&
+            instanceToRecords.has(oracleInstance) &&
+            instanceToRecords.get(oracleInstance) === records
+          ) {
+            // We should remove the entry entirely, as it is empty.
+            instanceToRecords.delete(oracleInstance);
+          }
+        },
+        async pushResult(roundId, result) {
+          // Sample of NaN, 0, or negative numbers get culled in
+          // the median calculation.
+          pushResult(roundId, result);
+        },
+      };
+
+      return harden(oracleAdmin);
+    },
+
+    /**
+     * consumers are encouraged to check
+     * that they're receiving fresh data by inspecting the updatedAt and
+     * answeredInRound return values.
+     * return is: [roundId, answer, startedAt, updatedAt, answeredInRound], where
+     * roundId is the round ID for which data was retrieved
+     * answer is the answer for the given round
+     * startedAt is the timestamp when the round was started. This is 0
+     * if the round hasn't been started yet.
+     * updatedAt is the timestamp when the round last was updated (i.e.
+     * answer was last computed)
+     * answeredInRound is the round ID of the round in which the answer
+     * was computed. answeredInRound may be smaller than roundId when the round
+     * timed out. answeredInRound is equal to roundId when the round didn't time out
+     * and was completed regularly.
+     *
+     * @param {bigint} _roundIdRaw
+     * @returns {Promise<{ roundId: bigint; answer: number; startedAt: bigint; updatedAt: bigint; answeredInRound: bigint; }>}
+     */
+    async getRoundData(_roundIdRaw) {
+      const roundId = Nat(_roundIdRaw);
+
+      assert(rounds.has(roundId), V3_NO_DATA_ERROR);
+
+      const r = rounds.get(roundId);
+
+      assert(r.answeredInRound > 0 && validRoundId(roundId), V3_NO_DATA_ERROR);
+
+      return {
+        roundId,
+        answer: r.answer,
+        startedAt: r.startedAt,
+        updatedAt: r.updatedAt,
+        answeredInRound: r.answeredInRound,
```

Comment:
In JS, patterns like this can simplify to:

```suggestion
        ...r,
        roundId,
```